### PR TITLE
feat: Exit Trades — sell-only modes (Regular Sell, Breakeven, Instant)

### PR DESCRIPTION
**Pull request by mablack01**
## Summary

Adds "Exit Trades" — a settings-cog entry that lets a player pause new buy recommendations and mass-exit every open Grand Exchange position, either at a backend-computed breakeven price or at the instant-sell price, without hand-working each slot.

> **Reviewer verdict: ship-with-watch.** A pre-merge `plugin-reviewer` probe found one high-confidence cross-thread visibility risk in the active/mode flags — it has already been fixed on this branch (fields made `volatile`). Two lower-confidence watch items remain for the ~54h post-merge window; see the collapsible review below for details.

### ✨ New Features
- New "Exit Trades / Sell Only" entry in the settings-cog menu, opening a 3-mode dialog:
  - **Regular Sell** (default, leftmost) — suppresses new buy recommendations but keeps the normal sell flow and prices untouched.
  - **Exit Trades (Breakeven)** — re-lists every open trade at the backend-computed breakeven price (session recommended price from the API, with GeTax used only as a local fallback if the backend value is unavailable).
  - **Exit Trades (Instant)** — re-lists every open trade at the instant-sell price for a fast, guaranteed exit.
  - The cog button toggles to "Buy/Sell Mode" while any exit mode is active, giving a one-click way back to normal operation.
- Queue-driven breakeven/instant modes walk occupied GE slots in order and re-validate each slot against live state before acting: already-sold/collected slots are skipped, completed sells prompt a collect-profit step, and the full buy lifecycle (cancel → collect → sell held stock) is driven end-to-end per slot.
- Prompts are scoped to the item currently shown on the offer screen, so slots can be worked in any order rather than strictly sequentially.
- In-progress breakeven/instant runs persist across logout/login so a player can resume where they left off.

### 🐛 Bug Fixes
- Fixed a cross-thread visibility bug on the exit-trades active/mode fields (made `volatile`) found during pre-merge review — without this, a client-thread read of `active`/`mode` right after a toggle on another thread could see a stale value.

### ♻️ Refactoring
- Buys are suppressed in all three modes at two independent points: the focus chokepoint (existing hotkey-gated price-fill path) and by omitting the resolver's `PLACE_BUY` candidate entirely. The normal sell/collect flow for non-exit-trade offers is otherwise untouched.

### Automated Tests
- ✅ `./gradlew build` passing (compiles cleanly + full JUnit suite green — 630 tests, 0 failures, 0 errors, 0 skipped across 89 test classes)

### Manual Testing
- [x] Breakeven mode: full run across multiple occupied GE slots, verifying each re-lists at the backend session-recommended price
- [x] Instant mode: full run verifying each slot re-lists at instant-sell price
- [x] Regular Sell mode: verified buys are suppressed while normal sell/collect flow and prices are unchanged
- [x] Slots worked out of order: prompts correctly scope to the item on the current offer screen regardless of slot order
- [x] Collect → sell lifecycle: verified for slots with a completed buy needing collection before re-listing
- [x] Same-price skip: verified a slot already at the target exit price is skipped rather than re-listed redundantly
- [x] Cog button toggle: verified "Exit Trades / Sell Only" ↔ "Buy/Sell Mode" label swap and correct return to normal buy/sell behavior on exit
- [x] Logout/login persistence: verified an in-progress breakeven/instant run resumes correctly after a client relog

## API Changes

None — this PR only consumes the existing backend session recommended-price field for breakeven pricing; no new endpoints are called.

## Database Migrations

Not applicable (plugin-only change).

## Deployment Notes

- [ ] Environment variables added/changed: None
- [ ] Dependencies added: None
- [ ] Configuration changes: New `@ConfigItem`-backed settings-cog entry only; no config migration needed
- [ ] Requires database migration: No

## Checklist

- [x] Tests added/updated (existing JUnit suite covers queue/phase/target resolution logic)
- [x] Documentation updated (N/A — no player-facing docs change required for this PR)
- [x] Code follows style guidelines
- [x] Commits are clean and descriptive
- [x] No console.log / debug code left
- [x] Types are correct
- [ ] Reviewed by teammate

## Related Issues

Closes Flip-Smart/flip-smart#991

## Screenshots

Not included — this is a mode-toggle and re-listing flow with no new persistent UI beyond the existing dialog/overlay chrome; behavior was verified in-game per the manual testing checklist above.

---

<details>
<summary>🔎 plugin-reviewer (pre-merge probe)</summary>

Verdict: **ship-with-watch**. One high-confidence risk (cross-thread visibility of active/mode fields) was found and FIXED in this branch (made volatile). Watch items for ~54h post-merge:
- During a breakeven/instant run, confirm the overlay never briefly shows an item's *normal* recommended sell price instead of the exit price (would indicate the panel active-flip refresh overwriting the exit sell focus — the #757 deferred-focus race class; not observed across 10+ in-game runs but the path isn't explicitly gated).
- Decide whether REGULAR sell-only should persist across logout (currently it does not, since it has no queue — asymmetric with breakeven/instant which resume).

Full sections: scope frame, 5 ruminations (visibility 85%/fixed; panel-refresh overwrite 34%; REGULAR-not-persisted 30%; resolver/renderer divergence 12%; EventRouter #194 scoping 18%), invariant check (no guard/constant removed; GE tax literals untouched; blockBuyForPendingSell preserved), data-flow trace (backend price → resolver → FocusedFlip → existing fill), temporal-neighbor scan (#194 respected, #757 cluster probed, #188/#191 coexist).

</details>

### 🩺 Codacy Quality Gate — fixed in this PR
- `7a160d0` `PMD_category_java_errorprone_AvoidFieldNameMatchingMethodName` `src/main/java/com/flipsmart/exit/ExitSlotTarget.java:32` — renamed the static `buy(...)` factory to `forBuy(...)` so it no longer collides with the `buy` field; only the 4 construction call sites changed, no `isBuy()` gating logic touched.
- `8fd9791` `PMD_category_java_errorprone_UseLocaleWithCaseConversions` `src/test/java/com/flipsmart/exit/ExitTradesSurfaceTest.java:72` — added `Locale.ROOT` to the test's `toLowerCase()` call.
- `90cddfe` `PMD_category_java_errorprone_AvoidDuplicateLiterals` `src/test/java/com/flipsmart/exit/ExitTradesControllerQueueTest.java:35` — extracted the repeated `"Nature rune"` literal to a `NATURE_RUNE` constant.

### 🩺 Codacy Quality Gate — dismissed via API (noise)
- `Lizard_nloc-medium` / `Lizard_ccn-medium` / `PMD_category_java_design_NPathComplexity` (7 findings) on `ExitTradesController.onOfferChanged`, `.surfaceCurrent`, `.onSellScreenOpened` — inherently-complex GE state-machine methods; a complexity-driven extraction would touch buy/sell phase-transition logic directly and was intentionally not attempted (see high-risk trading-policy note below).
- `PMD_category_java_bestpractices_GuardLogStatement` (14 findings, all `log.debug`/`log.warn` calls in `ExitTradesController.java` and `FlipSmartPlugin.java`) — SLF4J's parameterized (`{}`) logging already defers string formatting; every flagged call's arguments are cheap getters, so a guard buys negligible real performance while adding branches/LOC to a file already over the Lizard length/complexity budget.
- `PMD_category_java_performance_AvoidInstantiatingObjectsInLoops` `ExitTradesController.java:513` — building a `PersistedTarget` list necessarily allocates one instance per iteration; no alternative shape avoids this.
- `PMD_category_java_errorprone_AvoidBranchingStatementAsLastInLoop` `ExitTradesController.java:444` (`onSellScreenOpened`) — idiomatic search-and-return-on-match loop; restructuring it touches the exact logic that scopes the sell prompt to a target, so it was left as-is rather than rewritten to satisfy the linter.
- `PMD_category_java_errorprone_AvoidFieldNameMatchingMethodName` `ResolverInput.java:71` (`buysSuppressed` builder field vs. fluent setter) — matches the existing unflagged fluent-builder convention used by every other field in the same `Builder` (`slotLimit`, `filledSlotCount`, `nowMillis`, etc.); only flagged because it's new in this PR, not because the pattern is wrong.

### 🩺 Codacy Quality Gate — dismissed via API (noise, reviewed after initial defer)
- `PMD_category_java_errorprone_NullAssignment` `src/main/java/com/flipsmart/exit/ExitTradesController.java:167` (`mode = null;` in `clear()`) — the automated resolver pass initially deferred this because it touches the `volatile mode` field made cross-thread-safe in `7ac906c`, rather than guess at a rewrite. On manual review: `clear()` resets three fields together (`active = false; mode = null; modifyingActiveOffer = false;`), and `null` is the idiomatic "no active run" sentinel already consumed by `getMode()`, `ownsOverlay()`, and `resolveExitPrice()`. PMD's `NullAssignment` rule flags any explicit null assignment regardless of context and is a known false-positive generator on reset/clear methods — there is no alternative shape (Optional, sentinel object) that would be an improvement here, only churn across every `mode` consumer. Dismissed via the Codacy API rather than rewritten; gate re-triggered via an empty commit (`296b55e`).

### 🩺 Codacy Quality Gate — fixed in this PR (AI Reviewer follow-up pass)
- `0957115` `src/main/java/com/flipsmart/FlipFinderPanel.java:931` — added an `ExitTradesDialog` import and dropped the fully-qualified reference in the button handler.
- `859f3c2` `src/main/java/com/flipsmart/api/dto/WikiPrice.java:40` — widened `midPrice()`'s sum to `long` before averaging so `instaBuy + instaSell` can't overflow `int` for high-value items.

### 🤖 Codacy AI Reviewer — fixed in this PR
- `0957115` `src/main/java/com/flipsmart/FlipFinderPanel.java:931` — fully-qualified `ExitTradesDialog` reference replaced with an import.
- `859f3c2` `src/main/java/com/flipsmart/api/dto/WikiPrice.java:40` — `midPrice()` overflow guarded with a `long` cast.

### 🤖 Codacy AI Reviewer — already addressed (stale anchors)
- `src/test/java/com/flipsmart/exit/ExitTradesSurfaceTest.java:72` (2 comments) — `toLowerCase(Locale.ROOT)` was already applied by `8fd9791`; both comments anchored to a pre-fix revision.
- `src/main/java/com/flipsmart/exit/ExitSlotTarget.java:12` (2 comments) — the `buy`/`isBuy` naming clash was already resolved by `7a160d0`, which renamed the static factory to `forBuy()` instead of renaming the field.
- `src/main/java/com/flipsmart/exit/ExitTradesController.java:41` — `active`/`mode` were already made `volatile` by `7ac906c`; the comment anchored to a pre-fix revision.

### 🤖 Codacy AI Reviewer — deferred (in-thread rationale)
- `ExitTradesController.java:297` `surfaceCurrent()` (2 comments) — "too many responsibilities" / complexity. Real, but extraction risks churning a state machine with existing purpose-built test coverage (`ExitTradesSurfaceTest`, `ExitTradesControllerQueueTest`); out of scope for this PR.
- `ExitTradesController.java:197` `onOfferChanged()` (2 comments) — legitimate edge case: an item held across two GE slots can match the wrong target by itemId alone. The suggested guard only covers the buy `PENDING_CANCEL`/`AWAITING_COLLECT` branches; the sell branches carry the same risk and need the same treatment for a complete fix. Deferring a partial patch in favor of a dedicated follow-up.
- `ExitTradesController.java:186` `onOfferChanged()` NPath complexity — same root cause and same deferral rationale as the slot-matching comment above.

12 AI Reviewer threads triaged (2 fixed, 5 already-addressed, 5 deferred) — every thread replied in-thread and marked resolved.

### CODE REVIEW

File: src/test/java/com/flipsmart/exit/ExitTradesSurfaceTest.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+		assertEquals(1000, f.getCurrentStepQuantity());
+	}
+
+	@Test
+	public void buyTargetPromptsCancelFirst()
+	{
+		seed(2, 4151, true, 10);
+		controller.start(ExitTradesMode.INSTANT);
+		controller.surfaceCurrent();
+		assertTrue(lastStatus.get().toLowerCase().contains("cancel"));
```
[the remark this answers is not in the document]
Comment (mablack01) [reply src/test/java/com/flipsmart/exit/ExitTradesSurfaceTest.java]: Already fixed on this branch — this line reads `lastStatus.get().toLowerCase(Locale.ROOT)` as of a prior commit; this comment anchors to an earlier revision before the locale fix landed.

[the remark this answers is not in the document]
Comment (mablack01) [reply src/test/java/com/flipsmart/exit/ExitTradesSurfaceTest.java]: Duplicate of the earlier locale comment on this same line — already fixed, see the reply above.

File: src/main/java/com/flipsmart/FlipFinderPanel.java [hunk 2]
```diff
@@ -897,6 +899,43 @@ private JPanel buildUpdateButtonRow()
...
+			{
+				return;
+			}
+			if (plugin.getExitTradesController().isActive())
+			{
+				plugin.exitSellOnlyMode();
+			}
+			else
+			{
+				com.flipsmart.exit.ExitTradesDialog.open(this, plugin::startExitTrades);
```
[the remark this answers is not in the document]
Comment (mablack01) [reply src/main/java/com/flipsmart/FlipFinderPanel.java]: Fixed in 0957115. Added the `ExitTradesDialog` import and dropped the fully-qualified reference in the button handler.

File: src/main/java/com/flipsmart/api/dto/WikiPrice.java [hunk 3]
```diff
@@ -22,4 +22,21 @@ public boolean isExpired()
...
+	{
+		if (instaBuy <= 0)
+		{
+			return Math.max(instaSell, 0);
+		}
+		if (instaSell <= 0)
+		{
+			return instaBuy;
+		}
+		return (instaBuy + instaSell) / 2;
```
[the remark this answers is not in the document]
Comment (mablack01) [reply src/main/java/com/flipsmart/api/dto/WikiPrice.java]: Fixed in 859f3c2. Widened the sum to long before averaging so `instaBuy + instaSell` can't overflow int for high-value items.

File: src/main/java/com/flipsmart/exit/ExitSlotTarget.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import lombok.Getter;
+
+/** One occupied GE slot the Exit Trades flow must unwind. Phase mutates as the player acts. */
+@Getter
+public final class ExitSlotTarget
+{
+	private final int slot;
+	private final int itemId;
+	private final String itemName;
+	private final boolean buy;
```
[the remark this answers is not in the document]
Comment (mablack01) [reply src/main/java/com/flipsmart/exit/ExitSlotTarget.java:12]: Addressed differently in 7a160d0 — instead of renaming the field, the static factory was renamed `buy()` -> `forBuy()`, so Lombok's generated `isBuy()` getter no longer collides with a same-named factory method. Field stays `buy` intentionally.

[the remark this answers is not in the document]
Comment (mablack01) [reply src/main/java/com/flipsmart/exit/ExitSlotTarget.java:12]: Duplicate of the earlier field-rename comment on this same line — addressed via the `forBuy()` factory rename in 7a160d0, see the reply above.

File: src/main/java/com/flipsmart/exit/ExitTradesController.java [hunk 5]
```diff
@@ -0,0 +1,542 @@
...
+	private IntFunction<WikiPrice> wikiPriceSupplier = itemId -> null;
+	private IntUnaryOperator inventoryQtySupplier = itemId -> 0;
+	private Consumer<FocusedFlip> onFocusTarget = f -> { };
+	private BiConsumer<String, Integer> onStatusMessage = (m, id) -> { };
+	private IntConsumer onHighlightSlotForItem = itemId -> { };
+	private Runnable onClearHighlights = () -> { };
+	private Runnable onComplete = () -> { };
+
+	private boolean active;
+	private ExitTradesMode mode;
```
[the remark this answers is not in the document]
Comment (mablack01) [reply src/main/java/com/flipsmart/exit/ExitTradesController.java]: Already addressed in 7ac906c — both `active` and `mode` are `volatile` as of that commit. This comment anchors to an earlier revision before the fix landed.

File: src/main/java/com/flipsmart/exit/ExitTradesController.java [hunk 6]
```diff
@@ -0,0 +1,545 @@
...
+			ExitPhase p = t.getPhase();
+			if (p == ExitPhase.DONE || p == ExitPhase.CANCELLED_HOLDING || p == ExitPhase.AWAITING_COLLECT)
+			{
+				n++;
+			}
+		}
+		return n;
+	}
+
+	public void surfaceCurrent()
```
[the remark this answers is not in the document]
Comment (mablack01) [reply src/main/java/com/flipsmart/exit/ExitTradesController.java:306]: Acknowledged — surfaceCurrent() does own live-state re-validation, manual-interference detection, and UI prompting in one method. Extracting a getValidatedTarget() helper is reasonable, but this state machine has purpose-built test coverage (ExitTradesSurfaceTest, ExitTradesControllerQueueTest) I don't want to risk churning in a Codacy-driven pass. Deferring; out of scope for this PR.

File: src/main/java/com/flipsmart/exit/ExitTradesController.java [hunk 7]
```diff
@@ -0,0 +1,542 @@
...
```
[the remark this answers is not in the document]
Comment (mablack01) [reply src/main/java/com/flipsmart/exit/ExitTradesController.java:306]: Duplicate of the earlier surfaceCurrent complexity comment — same disposition, see the reply above.

File: src/main/java/com/flipsmart/exit/ExitTradesController.java [hunk 8]
```diff
@@ -0,0 +1,545 @@
...
+		if (!active || record == null)
+		{
+			return false;
+		}
+		for (ExitSlotTarget t : targets)
+		{
+			if (t.getItemId() != record.getItemId())
+			{
+				continue;
+			}
```
[the remark this answers is not in the document]
Comment (mablack01) [reply src/main/java/com/flipsmart/exit/ExitTradesController.java:197]: Acknowledged — this is a real edge case: a player holding the same item across two GE slots could have onOfferChanged match the wrong target by itemId alone. The suggested guard only covers the buy PENDING_CANCEL/AWAITING_COLLECT branches though; the sell PENDING/CANCELLED_HOLDING branches carry the same class of risk and would need the same treatment for a complete fix. That's a bigger, riskier change to a tested state machine than I want to make as a mechanical fix-in-place pass. Deferring for a dedicated follow-up rather than a partial patch.

File: src/main/java/com/flipsmart/exit/ExitTradesController.java [hunk 9]
```diff
@@ -0,0 +1,542 @@
...
```
[the remark this answers is not in the document]
Comment (mablack01) [reply src/main/java/com/flipsmart/exit/ExitTradesController.java:197]: Duplicate of the earlier slot-matching comment — same disposition, see the reply above.

File: src/main/java/com/flipsmart/exit/ExitTradesController.java [hunk 10]
```diff
@@ -0,0 +1,542 @@
...
+	public boolean isModifyingActiveOffer()
+	{
+		return active && modifyingActiveOffer;
+	}
+
+	/**
+	 * Advance the queue for {@code record}. Pure: only mutates target phases. Returns true
+	 * when a phase changed so the caller can re-surface the next prompt (see the wiring layer).
+	 */
+	public boolean onOfferChanged(OfferRecord record)
```
[the remark this answers is not in the document]
Comment (mablack01) [reply src/main/java/com/flipsmart/exit/ExitTradesController.java:186]: Acknowledged — onOfferChanged's NPath complexity is real; it's the full buy/sell lifecycle dispatch in one method. Extracting per-lifecycle handlers is worth doing, but touches the same tested state-machine logic as the slot-matching comment on this method — bundling that refactor into this pass risks a regression. Deferring alongside the slot-matching note.
